kubernetes/check.py

```diff
@@ -99,6 +101,10 @@ def __init__(self, name, init_config, agentConfig, instances=None):
         else:
             self._sd_backend = None
 
+        self.leader_candidate = agentConfig.get(LEADER_CANDIDATE)
```

Comment:
I'd move that option to `kubernetes.yaml` as that mechanism is k8s only and will be superseeded by the cluster level agent.
`kubernetes.yaml.example` should make it clear that users have three options:

- forcing event collection via `collect_events` = true
- going with the leader election method by setting `collect_events` = false and `event_leader_election` = true
- completely disabling events by setting both options to false